torch/csrc/jit/tensorexpr/cuda_codegen.cpp

```diff
@@ -620,11 +620,18 @@ void CudaCodeGen::CompileToNVRTC(
   // Note: hacked at::DeviceGuard since at::DeviceGuard was failing to work
   // properly in some scenarios
   const auto prior_device = at::cuda::current_device();
-  at::cuda::set_device(this->device().index());
+  if (prior_device != this->device().index()) {
+    at::cuda::set_device(this->device().index());
+  }
   // cudaSetDevice does not have to really change the underlying device if it
   // doesn't have to, so calling cudaFree to force that change
   CudaSetContext(pctx);
```

Comment:
Why keeping "CudaSetContext(pctx);", and the definition of "CudaSetContext"?

Reply:
Thanks for catching this. It must have crept in when I was moving to master. I am removing this.